v2/scripts/zebedee-content.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+DP_COMPOSE_V2_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+source "${DP_COMPOSE_V2_DIR}/scripts/utils.sh"
+
+REPO_DIR="$( cd "${DP_COMPOSE_V2_DIR}/../.." && pwd )"
+
+# Set VERBOSE=true to debug connection issues
+VERBOSE=${VERBOSE:-false}
+
+#==| CONSTANTS |================================================
+
+REPO=dp-zebedee-content
+REPO_URL=https://github.com/ONSdigital/$REPO
+REPO_CLONE_URL=git@github.com:ONSdigital/$REPO
+
+#==| FUNCTIONS |================================================
+
+clone_repo() {
+    # Test ssh to github
+    res=0
+    if [[ $VERBOSE = true ]]; then
+        ssh -v -T git@github.com || res=$?
+    else
+        ssh -T git@github.com &>/dev/null || res=$?
+    fi
+    case $res in
+        1)
+            info "github ssh authentication successful"
+            ;;
+        255)
+            fatal "github ssh authentication failed, ensure ssh is configured for github in order to use this script"
+            ;;
+        *)
+            fatal "github ssh authentication encountered unexpected error"
+    esac
+
+    local repo_pp="$(colour $CYAN $REPO)"
+
+    # Check if the repo already exists locally
+    local repo_path="$REPO_DIR/$REPO"
+    if [[ -d "${repo_path}" ]]; then
+        local branch=$(git -C "$repo_path" rev-parse --abbrev-ref HEAD)
+        local br_colour=$GREEN; [[ $branch == main ]] || br_colour=$YELLOW
+        local branch_pp=$(colour $br_colour $branch)
+        info "repo already cloned, skipping: $repo_pp ($branch_pp)"
+    else
+        # If not then clone it
+        info "cloning repo...: $repo_pp ($REPO_URL)"
+
+        if [[ $VERBOSE = true ]]; then
+            git -C "${REPO_DIR}" clone "$REPO_CLONE_URL"
+        else
+            git -C "${REPO_DIR}" clone "$REPO_CLONE_URL" 2> /dev/null
+        fi
+        if [[ $? > 0 ]]; then
+            fatal "failed to clone repo, please make sure you have access to it: $repo_pp ($REPO_URL)"
+        else
+            info "successfully cloned repo: $repo_pp ($REPO_URL)"
+        fi
+    fi
+}
```

Comment:
Aren't we using `clone_repo` elsewhere and could just call it?

Reply:
There is existing git clone logic, but no function it is buried in the clone.sh script so we can't just call it. We would need to factor the other git clone out to be more generic as the other is specifically for the cloning of repose with manifests (i.e. dockerised apps that form part of the compose stacks).

The git clone logic is a modified copy of that code. I chose to copy it rather than try to factor the other code out for two reasons:
1. The other clone implementation has additional logic that tailors it more to the specific use case of cloning the repos based on the manifest files and would need modification to make it generic enough to serve both use cases. This intern would make both scripts harder to read.
2. This is only the second use of the code and they have notable differences, so I thought it would be easier to maintain the two scripts if we copied the code across rather than factor out the git clone logic into separate functions in the utils library. The logic for testing the connection to github could easily be factored out so happy to do that as a compromise.